### PR TITLE
Tweak `10E0` for Vasco/ClimaRad HVAC

### PR DESCRIPTION
**Pull request by silverailscolo**
This PR adds signature information for the following hardware:

## Vasco
- `32:022222` HRU D60 (VMD-17RPS01) legacy product, no docs on Vasco site
- `37:117647` CO2 Sensor / Remote (VMS-17C01) [docs](https://vasco.eu/files/downloads/technical-sheet-rf-co2-switch/90.01.06.80-TECHNISCHE-FICHE-CO2-RF-BEDIENING-ENG.pdf)
- `29:091138` REM 4-button RF Remote (VMN-17LMP01) [docs](https://vasco.eu/files/downloads/technical-sheet-rf-3-position-switch/90.01.06.76-TECHNISCHE-FICHE-RF-SCHAKELAAR-ENG.pdf)

[Vasco D60 brochure NL.pdf](https://github.com/user-attachments/files/17402182/Vasco.D60.brochure.NL.pdf)

## ClimaRad
- `29:099029` FAN MiniBox (VMC-07RP01) [docs](https://www.climarad.nl/en/products/climarad-minibox/) 
- `37:153226` HRU Ventura V1x (VMD-07RPS13) [docs](https://www.climarad.nl/en/products/climarad-ventura-v1x/)
- `29:123150` REM 4-button RF Remote (VMN-07LM01) [docs](https://www.climarad.nl/en/products/climarad-wireless-control-panel-rf/)

### CODE REVIEW

File: src/ramses_tx/parsers.py [hunk 1]
```diff
@@ -750,7 +751,7 @@ def parser_0418(payload: str, msg: Message) -> PayDictT._0418 | PayDictT._0418_N
         return null_result
 
     # NOTE: such payloads have idx=="00": if verb is I, can safely assume log_idx is 0,
-    # but for RP it is sentinel for null (we can't know the correspondings RQ's log_idx)
+    # but for RP it is sentinel for null (we can't know the corresponding RQ's log_idx)
```
Comment (zxdavb) [inline src/ramses_tx/parsers.py]: Do continue to point out any typos, etc. - but they should be separate from your PR - I will fix this & you can rebase your PR.

File: src/ramses_tx/fingerprints.py [hunk 2]
```diff
@@ -43,6 +43,7 @@
     "0001001B371B01FEFF": ("FAN", "37", "2019-08-29", "CVE-RF"),  # .                       31D9, 31DA
     "0001001B381B01FEFF": ("FAN", "37", "2020-02-14", "CVE-RF"),  # .                       31D9, 31DA (and I|042F, I|3120)
     "0001001B391B01FEFF": ("FAN", "37", "2021-11-04", "CVE-RF"),
+    "0001C8830C0A65FEFF": ("FAN", "37", "2020-12-17'", "VMD-07RPS13"), # .                   ClimaRad Ventura
```
Comment (zxdavb) [inline src/ramses_tx/fingerprints.py]: Please provide the corresponding `10E0` packet in:
 - **tests/tests/parsers/code_10e0.log**
 
 Also for the other signatures, below.

File: src/ramses_tx/schemas.py [hunk 3]
```diff
@@ -232,7 +232,7 @@ def convert_null_to_dict(node_value: _T | None) -> _T | dict[Never, Never]:
 SCH_DEVICE_ID_UFC = vol.Match(DEVICE_ID_REGEX.UFC)
 
 _SCH_TRAITS_DOMAINS = ("heat", "hvac")
-_SCH_TRAITS_HVAC_SCHEMES = ("itho", "nuaire", "orcon")
+_SCH_TRAITS_HVAC_SCHEMES = ("climarad", "itho", "nuaire", "orcon", "vasco")
```
Comment (zxdavb) [inline src/ramses_tx/schemas.py]: This is not part of this PR, it should be with `22F1`: please remove it from this PR.

BTW, climarad / vasco use the same scheme? If so, call it vasco.

File: src/ramses_tx/parsers.py [hunk 4]
```diff
@@ -1653,6 +1654,7 @@ def parser_22f3(payload: str, msg: Message) -> dict[str, Any]:
     }.get(int(payload[2:4], 0x10) & 0xC0)  # 0b1100-0000
 
     duration = int(payload[4:6], 16) * 60 if units == "hours" else int(payload[4:6], 16)
+    result = {}
```
Comment (zxdavb) [inline src/ramses_tx/parsers.py]: This should be in a separate PR, for `22F3`.

Please remove it from this PR.

**Comment by zxdavb** [comment]
I recommend you put this PR on hold, and do `22F1` first. 

Experience tells me that will be the easiest way forward.

**Review by zxdavb — CHANGES_REQUESTED** [review]

File: src/ramses_tx/schemas.py (hunk 3 above)
Comment (silverailscolo) [reply src/ramses_tx/schemas.py]: ClimaRad uses the vasco scheme on their MiniBox fan, but a different one on the Ventura HRU (via 22F4). Will split them up

Comment (zxdavb) [reply src/ramses_tx/schemas.py]: As far as I can recall (and I wrote most of the code), schemes only apply to `22F1`.

I have no idea what a `22F4` is for - I'd love your input on that.

Comment (silverailscolo) [reply src/ramses_tx/schemas.py]: ClimaRad doesn't share their list, but their VenturaV1x HRU uses 22F4 for speed steps, while others use 22F1.
So our code needs a scheme for that, and I would expect that scheme (as well as this list of scheme names) to live close to the existing ones. But I will wait till I work on 22F4.

File: tests/tests/parsers/code_10e0.log [hunk 5]
```diff
@@ -99,10 +99,16 @@
 2023-10-26T00:00:00.000000 ...  I --- 37:211856 --:------ 37:211856 10E0 038 000001001B391B01FEFFFFFFFFFF040B07E54356452D52460000000000000000000000000000  # {'description': 'CVE-RF',              'manufacturer_sub_id': '00', 'product_id': '1B', 'date_2': '0000-00-00', 'date_1': '2021-11-04', 'oem_code': '01'}  # Manufacturer is Buva?
 2022-11-03T22:50:34.060486 ...  I --- 37:255297 63:262142 --:------ 10E0 038 00000100290A0201FFFFFFFFFFFF140407E5564D532D31324842313600000000000000000000  # {'description': 'VMS-12HB16',          'manufacturer_sub_id': '00', 'product_id': '29', 'date_2': '0000-00-00', 'date_1': '2021-04-20', 'oem_code': '01'}  #
 
-# oem_code '66'  - Vasco CH/DHW & HVAC
+# oem_code '65' - ClimaRad HVAC
+2024-10-09T11:50:24.287879 123  I --- 37:153226 63:262142 --:------ 10E0 038 000001C8830C0A65FEFFFFFFFFFF110C07E4564D442D30375250533133000000000000000000  # {'description': 'VMD-07RPS13',         'manufacturer_sub_id': 'C8', 'product_id': '83', 'date_1': '2020-12-17', 'date_2': '0000-00-00', 'oem_code': '65'}  # ClimaRad VenturaV1x
```
Comment (zxdavb) [inline tests/tests/parsers/code_10e0.log:103]: Add this to **tests/tests/fingerprints/hvac/37_VMD-07RPS13.log**

File: tests/tests/parsers/code_10e0.log [hunk 6]
```diff
@@ -99,10 +99,16 @@
...
+
+# oem_code '66' - Vasco CH/DHW & HVAC
 2023-02-19T11:37:59.549235 072 RP --- 29:079817 30:006056 --:------ 10E0 029 000001C81C090466FEFFFFFFFFFFFFFFFFFF564D432D31375250303100                    # {'description': 'VMC-17RP01',          'manufacturer_sub_id': 'C8', 'product_id': '1C', 'date_2': '0000-00-00', 'date_1': '0000-00-00', 'oem_code': '66'}  # Vasco C400RF extraction unit (EXT, not HRU/PIV)
 2023-02-18T19:56:43.537735 059  I --- 29:136571 63:262142 --:------ 10E0 038 000001C822030166FEFFFFFFFFFF070507DF564D532D31374330310000000000000000000000  # {'description': 'VMS-17C01',           'manufacturer_sub_id': 'C8', 'product_id': '22', 'date_2': '0000-00-00', 'date_1': '2015-05-07', 'oem_code': '66'}  # CO2 sensor
+2024-10-09T11:51:49.472184 059  I --- 37:117647 63:262142 --:------ 10E0 038 000001C822060166FEFFFFFFFFFF160C07E0564D532D31374330310000000000000000000000  # {'description': 'VMS-17C01',           'manufacturer_sub_id': 'C8', 'product_id': '22', 'date_1': '2016-12-22', 'date_2': '0000-00-00', 'oem_code': '66'}  # CO2 sensor
```
Comment (zxdavb) [inline tests/tests/parsers/code_10e0.log:110]: Add this to **tests/tests/fingerprints/hvac/99_VMS-17C01.log**

File: tests/tests/parsers/code_10e0.log [hunk 7]
```diff
@@ -99,10 +99,16 @@
...
+2024-10-04 22:24:58.581000 ...  I --- 29:091138 63:262142 --:------ 10E0 038 000001C8400F0166FFFFFFFFFFFF0E0207E3564D4E2D31374C4D503031000000000000000000  # {'description': 'VMN-17LMP01',         'manufacturer_sub_id': 'C8', 'product_id': '40', 'date_1': '2019-02-14', 'date_2': '0000-00-00', 'oem_code': '66'}  # Vasco 4-button remote
```
Comment (zxdavb) [inline tests/tests/parsers/code_10e0.log:111]: Add this to **tests/tests/fingerprints/hvac/29_VMN-17LMP01.log**

**Comment by zxdavb** [comment]
If you could list the make/model of the devices that your adding `10E0` packets for? A link to a URL of that product would be useful too.

**Review by zxdavb — CHANGES_REQUESTED** [review]

File: tests/tests/parsers/code_10e0.log (hunk 5 above)
Comment (silverailscolo) [reply tests/tests/parsers/code_10e0.log:103]: You mean add (copy) or move (as remove here)?

Comment (zxdavb) [reply tests/tests/parsers/code_10e0.log:103]: a duplicate copy (not a move)

**Comment by silverailscolo** [comment]
> If you could list the make/model of the devices that your adding `10E0` packets for? A link to a URL of that product would be useful too.

Where would you want this information to be stored? Don't see any urls in the 10E0 or fingerprints/hvac yet

File: tests/tests/parsers/code_10e0.log (hunk 7 above)
Comment (silverailscolo) [reply tests/tests/parsers/code_10e0.log:111]: Done

**Comment by zxdavb** [comment]
> Where would you want this information to be stored?

See first post, which I have edited.

File: tests/tests/fingerprints/hvac/37_VMD-17RPS13.log [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# ClimaRad Ventura V1x HRU
+
+2024-10-09T11:50:24.287879 123  I --- 37:153226 63:262142 --:------ 10E0 038 000001C8830C0A65FEFFFFFFFFFF110C07E4564D442D30375250533133000000000000000000  # {'description': 'VMD-07RPS13',         'manufacturer_sub_id': 'C8', 'product_id': '83', 'date_1': '2020-12-17', 'date_2': '0000-00-00', 'oem_code': '65'}  # ClimaRad VenturaV1x
```
Comment (zxdavb) [inline tests/tests/fingerprints/hvac/37_VMD-17RPS13.log]: This file should be called **37_VMD-07RPS13.log**.

File: tests/tests/fingerprints/hvac/29_VMN-17LMP01.log [hunk 9]
```diff
@@ -0,0 +1,4 @@
+# ClimaRad MiniBox Fan
+2024-10-16T18:43:52.761437 087  I --- 29:099029 63:262142 --:------ 10E0 029 000001C81D150765FFFFFFFFFFFFFFFFFFFF564D432D30375250303100  # {}
```
Comment (zxdavb) [inline tests/tests/fingerprints/hvac/29_VMN-17LMP01.log]: This should have a dict after the `#`

Comment (zxdavb) [reply tests/tests/fingerprints/hvac/29_VMN-17LMP01.log]: ... and is it a **VMC-07RP01**?

**Review by zxdavb — CHANGES_REQUESTED** [review]

**Comment by zxdavb** [comment]
Nice one!
